### ISSUE TITLE
[RFC] Deprecating and removing experimental.set_resource() API

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#13021]
See: https://github.com/ray-project/ray/pull/13019

This API is currently planned for removal since it is buggy and not implemented in the new scheduler refactor. Please comment here if you have use cases that depend on the set resource API that cannot be ported to placement groups.

**Comment by stephanie-wang** [comment]
I'm using experimental.set_resource for a benchmark that tests Ray's recovery mechanism.

I'd prefer to use placement groups actually, but the current semantics are not sufficient. I use experimental resources for two reasons:
1. To place a set of parallel tasks at the beginning of the job, using a STRICT_SPREAD-like strategy.
2. Halfway during the job, my script kills and restarts one of the worker raylets. The driver waits for the raylet to restart and then adds the custom resource back to the new raylet, so that the failed tasks can run again.

It looks like the current placement groups semantics would fulfill the first purpose, but not the second. According to the docs, if one bundle in the placement group is lost due to node failure, *all* bundles will get rescheduled. I don't want this to happen because it will affect tasks that can still run.

**Comment by ericl** [comment]
> According to the docs, if one bundle in the placement group is lost due to node failure, all bundles will get rescheduled.

@rkooo567 is this the case? I think the doc should say "reschedule all lost bundles" right?

> The driver waits for the raylet to restart and then adds the custom resource back to the new raylet

Maybe I'm missing something here, but is it possible to add the custom resource to the ray start flags? (or, can you use the existing `node_id:<ip>` resource?)

**Comment by rkooo567** [comment]
Yes it is rescheduling the lost bundles. I will fix the doc if it wasn’t specified correctly

**Comment by stephanie-wang** [comment]
Oh I see, maybe it's okay then :) I interpreted this quote from the docs to mean that it would reschedule all bundles:

```
If nodes that contain some bundles of a placement group die, all the bundles will be rescheduled on different nodes by GCS.
```

**Comment by PidgeyBE** [comment]
We use it for several purposes. Most of them can indeed be replaced with placement groups or be refactored out of our codebase.
The only one that can not be replaced AFAIK is:
* We set the GPU type as a custom resource on nodes, e.g. `GPU:GeForce RTX 2080 SUPER: 3`. Meanwhile the `accelerator_type:RTX` custom resource has been added by ray core, but it is less detailed.

**Comment by ericl** [comment]
@PidgeyBE can you use static resources for that (pass --resources to ray start)?

cc @wuisawesome if we can get more detailed accelerator type, not sure if "RTX" was the intended detected gpu type here.

**Comment by PidgeyBE** [comment]
@ericl That's only possible if you know at deployment time what kind of GPU your worker is going to get. We have a few setups where that's not the case. Some customers have a PC with e.g. 1 strong/expensive GPU (for training ML models) and 1 cheaper (for doing inference). We use kubernetes and deploy 2 ray worker pods, 1 pod gets the expensive GPU, the other one the cheaper. But it's kubernetes that decides which ray worker gets which GPU, and it's pretty random afaik.
But we could make a start up script, that scrapes `nvidia-smi` and passes a key into `ray start --resources`. 
So there is a workaround indeed :)

**Comment by ericl** [comment]
@PidgeyBE the startup script sounds like it should work. You can also add resources the `resources` field of the multi node type autoscaling config and those will be added automatically.

**Comment by wuisawesome** [comment]
Closed by #13019
